tests/foreman/api/test_environment.py

```diff
@@ -30,6 +30,83 @@
 from robottelo.utils.datafactory import valid_environments_list
 
 
+@pytest.mark.e2e
+@pytest.mark.tier1
+@pytest.mark.upgrade
+def test_positive_CRUD_with_attributes(
+    session_puppet_enabled_sat, module_puppet_org, module_puppet_loc
+):
+    """Check if Environment with attributes can be created, updated and removed
+
+    :id: d2187971-86b2-40c9-a93c-66f37691ae2c
+
+    :BZ: 1337947
+
+    :expectedresults:
+        1. Environment is created and has parameters assigned
+        2. Environment can be listed by parameters
+        3. Environment can be updated
+        4. Environment can be removed
+    """
+    puppet_sat = session_puppet_enabled_sat
+    # Create with attributes
+    env_name = gen_string('alpha')
+    environment = puppet_sat.api.Environment(
+        name=env_name, organization=[module_puppet_org], location=[module_puppet_loc]
+    ).create()
+    assert env_name == environment.name
+    assert len(environment.organization) == 1
+    assert environment.organization[0].id == module_puppet_org.id
+    assert len(environment.location) == 1
+    assert environment.location[0].id == module_puppet_loc.id
+
```

Comment:
Should we also assert the name here keeping in mind the search query we are using below?
```suggestion
assert environment.location[0].name ==module_puppet_loc.name
assert environment.organization[0].name ==module_puppet_org.name

```

Reply:
I guess this wouldn’t add much value, and you said search query below is already using it so its covered already